tests/recipes/recipe_lib_test.py

```diff
@@ -0,0 +1,159 @@
+        mock_glob.assert_called()
+        mock_ensure_dir.assert_called()
+        mock_find_executable.assert_called()
+        mock_check_recipe_choices.assert_called()
+
+    @mock.patch("pythonforandroid.util.chdir")
+    @mock.patch("pythonforandroid.build.ensure_dir")
+    @mock.patch("pythonforandroid.archs.glob")
+    @mock.patch("pythonforandroid.archs.find_executable")
+    def test_build_arch(
+        self,
+        mock_find_executable,
+        mock_glob,
+        mock_ensure_dir,
+        mock_current_directory,
+    ):
+        mock_find_executable.return_value = self.expected_compiler.format(
+                android_ndk=self.ctx._ndk_dir
+        )
+        mock_glob.return_value = ["llvm"]
+
+        # Since the following mocks are dynamic,
+        # we mock/test it inside a Context Manager
+        with mock.patch(
+            f"pythonforandroid.recipes.{self.recipe_name}.sh.Command"
+        ) as mock_sh_command, mock.patch(
+            f"pythonforandroid.recipes.{self.recipe_name}.sh.make"
+        ) as mock_make:
+            self.recipe.build_arch(self.arch)
+
+            for command in self.sh_command_calls:
+                self.assertIn(
+                    mock.call(command),
+                    mock_sh_command.mock_calls,
+                )
+            mock_make.assert_called()
+
+        # make sure that the mocked methods are actually called
+        mock_glob.assert_called()
+        mock_ensure_dir.assert_called()
+        mock_current_directory.assert_called()
+        mock_find_executable.assert_called()
+
+
+class BaseTestForCmakeRecipe(BaseTestForMakeRecipe):
+    """
+    An unittest for testing any recipe using `cmake`. It inherits from
+    `BaseTestForMakeRecipe` but we override the build method to match the cmake
+    build method.
+
+    .. note:: Note that Some cmake recipe may need some more specific testing
+        ...but this should cover the basics.
+    """
+
+    @mock.patch("pythonforandroid.util.chdir")
+    @mock.patch("pythonforandroid.build.ensure_dir")
+    @mock.patch("pythonforandroid.archs.glob")
+    @mock.patch("pythonforandroid.archs.find_executable")
+    def test_build_arch(
+        self,
+        mock_find_executable,
+        mock_glob,
+        mock_ensure_dir,
+        mock_current_directory,
+    ):
+        mock_find_executable.return_value = self.expected_compiler.format(
+                android_ndk=self.ctx._ndk_dir
+        )
+        mock_glob.return_value = ["llvm"]
+
+        # Since the following mocks are dynamic,
+        # we mock/test it inside a Context Manager
+        with mock.patch(
+            f"pythonforandroid.recipes.{self.recipe_name}.sh.make"
+        ) as mock_make, mock.patch(
+            f"pythonforandroid.recipes.{self.recipe_name}.sh.cmake"
+        ) as mock_cmake:
+            self.recipe.build_arch(self.arch)
+
+            mock_cmake.assert_called()
+            mock_make.assert_called()
```

Comment:
Same here it could also be outside. I know it can feel weird at first because we assume the variable is scoped to the that bock, but not. Unlike javascript `let` this Python variable is function scoped, just like javascript `var`.

Reply:
Thanks man, you are right :smile:, let me change this and try if a `force-push` with a rebase on top of develop branch triggers the build...